sxg_rs/src/process_html.rs

```diff
@@ -0,0 +1,241 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::http::HttpResponse;
+use crate::http_parser::link::Link;
+use crate::link::ALLOWED_PARAM_NAMES;
+use anyhow::Result;
+use once_cell::sync::Lazy;
+use regex::Regex;
+use serde::Deserialize;
+use std::borrow::Cow;
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ProcessHtmlOption {
+    is_sxg: bool,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+enum ContentType {
+    HtmlUtf8,
+    HtmlOther,
+    Other,
+}
+fn parse_content_type(content_type_header_value: &str) -> ContentType {
+    static CONTENT_TYPE_HTML: Lazy<Regex> =
+        Lazy::new(|| Regex::new(r#"^text/html([ \t]*;[ \t]*charset=(utf-8|"utf-8"))?$"#).unwrap());
```

Comment:
Make it case-insensitive, e.g. by adding `(?i)` to the front of the regex.

Reply:
Done